File: android/app/src/main/java/com/example/mywebview/urb/core/UrbCommandRegistry.kt
package com.example.mywebview.urb.core

class UrbCommandRegistry {
    private val commands = mutableMapOf<String, UrbCommand>()

    fun register(command: UrbCommand): UrbCommandRegistry {
        commands[command.name] = command
        return this
    }

    fun find(name: String): UrbCommand? = commands[name]
}

File: ios/App/URB/Core/UrbCommandRegistry.swift
import Foundation

final class UrbCommandRegistry {
    private var commands: [String: UrbCommand] = [:]

    @discardableResult
    func register(_ command: UrbCommand) -> UrbCommandRegistry {
        commands[command.name] = command
        return self
    }

    func find(_ name: String) -> UrbCommand? {
        commands[name]
    }
}
